File: 프로그래머스/1/132267. 콜라 문제/콜라 문제.java
class Solution {
    public int solution(int a, int b, int n) {
        int answer = 0;

        while (n / a != 0) {
            
            int temp = n;

            n = n / a * b;
            answer += n;
            n += temp % a;
        }

        return answer;
    }
}

File: Kotlin/프로그래머스/1/132267. 콜라 문제/콜라 문제.kt
class Solution {
    fun solution(a: Int, b: Int, n: Int): Int {
        var answer: Int = 0

        var original = n
        var left = 0

        //2 1 20
        while (original >= a) {
            val add = original / a * b
            answer += add
            original %= a
            original += add

        }

        return answer
    }
}